prophet-ressa/src/gen_ressa.rs

```diff
@@ -0,0 +1,96 @@
+use prophet_ressa_minify::try_minify_ressa;
+use source_code_parser::ressa::NodePattern;
+use source_code_parser::{ressa::Indexable, Language, ModuleComponent};
+use std::collections::HashSet;
+use std::fs::DirEntry;
+
+use crate::Error;
+
+/// Use the langages in the provided LAAST to determine and load the needed ReSSAs
+pub fn extract_ressas(
+    ast: &Vec<ModuleComponent>,
+    ressa_dir: &str,
+) -> Result<Vec<NodePattern>, Error> {
+    // Get subdirectories
+    let dirs = get_subdirs(ressa_dir)?;
+
+    // Find languages
+    let langs = ast
+        .iter()
+        .flat_map(|module| find_languages(module as &dyn Indexable))
+        .collect::<HashSet<Language>>();
+
+    // Create ReSSAs for the languages in the project
+    let mut ressas = vec![];
+    let minified_ressas = dirs
+        .into_iter()
+        .flat_map(|entry| {
+            let name = entry.file_name();
+            let name = name.to_str()?;
+            let lang = to_lang(name)?;
+            Some((name.to_string(), entry.path().is_dir(), lang))
+        })
```

Comment:
For some reason I thought for the longest time you couldn't use `?` in closures and I'm not sure where I got that idea from. Glad I know now that you can.